### DISCUSSION TITLE
Unable to download Diamond on Windows

### DISCUSSION THREAD

**Discussion by bonaventuredossou** [bbuchfink/diamond#494] [Q&A] [answered]
I am unable to install diamond on Windows neither can I through "pip install ....". I went through the documentation, which does not seem to help.

How can I proceed, please?

Thanks in advance

**Comment by bbuchfink** [comment, marked as answer]
You can't install diamond by pip, go to releases (https://github.com/bbuchfink/diamond/releases) and download the Windows package.

**Reply by bonaventuredossou** [reply]
I did try that - unfortunately on W10 it does not even launch

**Reply by bbuchfink** [reply]
It's not a graphical application. You need to go to the command prompt (cmd.exe) and run it from there.

**Reply by bonaventuredossou** [reply]
I just tried many - still no fortune. After the command "start diamond.exe", nothing gets launched

**Reply by dtabb73** [reply]
I am experiencing the same thing on my Windows 10 machine.  Whether I enter "diamond" by itself or "diamond help" or "diamond makedb..." the response is always a single blank line of output and then the command prompt returns.  The Linux version works just fine on my other computer, though.  The machine on which it is failing does have an older CPU, though; it's a Xeon X5670.  Is it possible that this binary assumes that this CPU supports AVX instructions?

**Reply by bbuchfink** [reply]
Yes, that may be possible. I'll try to compile something that works on older CPUs.

**Reply by dtabb73** [reply]
Thank you!  I will be teaching my demo on my laptop, and happily, it has a later-generation CPU that supports AVX.  The binary works fine over there.

**Reply by bbuchfink** [reply]
Could you let me know if this binary works on your older system?

[diamond.zip](https://github.com/bbuchfink/diamond/files/8590828/diamond.zip)

**Reply by bbuchfink** [reply]
Hi @dtabb73, would it be possible that you test this new binary on your older Windows system: https://github.com/bbuchfink/diamond/releases/download/v2.1.6/diamond-windows.zip

I have recently done some work to improve compatibility but don't have an older Windows system to test this myself.

**Reply by dtabb73** [reply]
Hi, Benjamin.

I have moved home to South Africa, leaving that older computer behind in France, but I believe I could test this on a "Sandy Bridge" processor system here at home.  It might take me a couple of days to enable its use.  I will try to let you know soon!

Dave

**Reply by dtabb73** [reply]
Hi, @bbuchfink. 

Over the weekend, I was able to download the diamond source to my old Mac Mini (using an i5-2415 processor under macOS 10.13.6).  I installed Xcode and CMake.  Then I was able to build diamond successfully on the first try.  I used makedb to index the MYCTU reference proteome from UniProt and then queried it with an M. smegmatis protein.  I had my ortholog in less than a second.

I moved next to my old Lenovo ThinkCentre, running an i5-2400 CPU under Windows 10.  This time I downloaded the Windows executables rather than starting from source.  My first attempt to run diamond revealed that I was missing a library (vcruntime140_1.dll).  I installed the 64-bit Visual Studio redistributable, and then the software ran fine.  Again I queried an M. smegmatis protein against the UniProt reference proteome for MYCTU.  The search reported back the ortholog in no time flat.

Thanks for keeping older computers in mind for homology search.

**Reply by bbuchfink** [reply]
Thanks for testing this and happy to hear that it's working!

**Comment by dtabb73** [comment]
Hi, Benjamin.

Thank you very much.  It worked perfectly!

I filmed a short screen capture video to explain how to use it to newcomers.  Hopefully I haven't misinterpreted your paper too much.  It appears here: https://www.youtube.com/watch?v=BDOSyrKy-UA.

Thanks,
Dave
